packages/next/src/server/web/spec-extension/response.ts

```diff
@@ -26,10 +26,11 @@ function handleMiddlewareField(
   }
 }
 
-export class NextResponse extends Response {
+export class NextResponse<Body = unknown> extends Response {
   [INTERNALS]: {
     cookies: ResponseCookies
     url?: NextURL
+    Body?: Body
```

Comment:
This seems redundant?

Reply:
Yeah that's what I thought at first too, but `[INTERNALS]` is apparently important to allow for checking against the consequent executions of the `.json()` method, check out the line with the ❌ below (it should be an error but it is not)

[TypeScript Playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAkgOQCoFEBKCCCAZAyjAXhlwE8BbAIxABsAKAcgEswoBTAJzAENqZ3WIAB3ARW9AJQAoSawAew9rGDUuECDARyoaAcMisAPACEQAExKEYAVzABrMCADuYAHwwtrMKfU6hI1jAA3pIwMADaiKgYOLgAugBcQSGhMAD0qTAm5gD8iVkkADTJAL7SoaCQUOxWwFAg7LRUOXlmJHBgjLAAPjBgVtTUBTDMnYm+eqLtnZaBxeJJKTAQVoIcja1DI1BSyaFQABaMEBHI6Fh4sTOloaXJ0FxQjMAwAFYQ4AYAUu9g+S7r5kS33A+U2HSguRg438U22iU0sm0un8Xx+fwWKXSMAAAlAIABaOSrWqE9jseowJCHdRHXogWDMGDURgUVKmEBkPiMADm+1gDkcQy4XhgjgCwGF9FgXCs3LIngZUAAdLsYBVoHxkfoxlrRJZofolW9wADCsNwVJFvwoFZOL1WI4NFoDaJaPw-Iammb3RNWJaYKVbkT6rAAGY2WqMcAwADiKCQtHE8OdusMgS4FGAiQAjAAmADMxTcgTSGRjng4TxgXHY3Ks8pYMEEqlEpmSjFDMFoAFkHvslexheyyImYG4AAxKgCs8xLquttrATsRLtYRp+tHTmZzBbmpZggFByOnuMn1RJetX1fi1ErJBd2hFIj2idcmwKmVihxL0WS-+h7rFABlyY8OHJdhz1YCUrD1EBOzII4IGYblwkiM4YliJUvUkIN5BDGBwzASNozjJBc0TZMV1TIJ9wQEAYG5Ct2CrGs6wbWBoNYNtQg7LtewOAchw5UcJ2nWcYHnVgbQfFNnzXY0wE3DMsxgPNC3mLEjwcE8wMSYULAvLhgGAVhBDYLiAzvSTF2XJ9fVfBT30-b9f1kf91IyTS6NAs9qzAfTWmrIyTLM7DJCAA)

![Screenshot 2023-05-17 at 22 27 05](https://github.com/vercel/next.js/assets/1935696/bec5e993-ad3f-40c1-8bf6-98cb3890d8a7)

Reply:
Should that be `body?: Body` then (lowercased)?

Reply:
Done in 3f59852a7476fa1c2647145b68109976dbc30853